diff --git a/libfprint/drivers/goodixmoc/goodix.c b/libfprint/drivers/goodixmoc/goodix.c
index 1d8ea0b..7a3d195 100644
--- a/libfprint/drivers/goodixmoc/goodix.c
+++ b/libfprint/drivers/goodixmoc/goodix.c
@@ -1423,6 +1423,7 @@ gx_fp_probe (FpDevice *device)
     case 0x659A:
     case 0x6890:
     case 0x6984:
+    case 0x6F94:
       self->max_enroll_stage = 12;
       break;
 
@@ -1688,6 +1689,7 @@ static const FpIdEntry id_table[] = {
   { .vid = 0x27c6,  .pid = 0x689A,  },
   { .vid = 0x27c6,  .pid = 0x66A9,  },
   { .vid = 0x27c6,  .pid = 0x6984,  },
+  { .vid = 0x27c6,  .pid = 0x6F94,  },
   { .vid = 0,  .pid = 0,  .driver_data = 0 },   /* terminating entry */
 };
 
